numberOfProvinces.js

```diff
@@ -0,0 +1,115 @@
+/*
+n cities
+
+a - b - c
+
+i: n x n matrix
+o: total number of provinces
+
+directly or indirectly
+
+Input: isConnected = [[1,1,0],[1,1,0],[0,0,1]]
+Output: 2
+
+   0 1 2
+0 [1,1,0],
+1 [1,1,0],
+2 [0,0,1]
+
+*/
+
+const { Queue } = require('./utils/queue');
+
+const directions = [
+    [1, 0],
+    [0, 1],
+    [-1, 0],
+    [0, -1],
+];
+
+const CONNECTED = 1;
+
+const getRowColString = (row, col) => `${row}, ${col}`;
+
+const isInBounds = (row, col, adjacencyMatrix) => {
+    const numRows = adjacencyMatrix.length;
+    const numCols = adjacencyMatrix[0].length;
+
+    const isInRowBound = row >= 0 && row < numRows;
+    const isInColBound = col >= 0 && col < numCols;
+
+    return isInRowBound && isInColBound;
+};
+
+const getNeighbors = (row, col, adjacencyMatrix) => {
+    const neighbors = [];
+
+    for (const direction of directions) {
+        const [rowChange, colChange] = direction;
+
+        const newRow = row + rowChange;
+        const newCol = col + colChange;
+
+        const isInBound = isInBounds(newRow, newCol, adjacencyMatrix);
```

Comment:
Having a variable and function with nearly identical names can be a bit confusing. I would try to avoid this. In this case, I would just write `if (!isInBounds(newRow, newCol, adjacencyMatrix))`